app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheet.kt

```diff
@@ -108,12 +130,47 @@ class BloodPressureEntrySheet : BottomSheetActivity(), BloodPressureEntryUi {
             dayTextChanges(),
             monthTextChanges(),
             yearTextChanges()
-        ),
+        )
+        .compose(ReportAnalyticsEvents())
+        .share()
+  }
+
+  @SuppressLint("CheckResult")
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    setContentView(R.layout.sheet_blood_pressure_entry)
+    TheActivity.component.inject(this)
+
+    if (savedInstanceState != null) {
+      delegate.onRestoreInstanceState(savedInstanceState)
+    } else {
+      delegate.prepare()
+    }
+
+    bindUiToControllerWithoutDelay(
+        ui = this,
+        events = events,
         controller = controller,
         screenDestroys = screenDestroys
     )
   }
 
+  override fun onStart() {
+    super.onStart()
+    delegate.start()
+  }
+
+  override fun onStop() {
+    delegate.stop()
+    super.onStop()
+  }
+
+  override fun onSaveInstanceState(outState: Bundle?) {
+    delegate.onSaveInstanceState(outState)
+    super.onSaveInstanceState(outState)
```

Comment:
Should this be the state that is returned from `MobiusDelegate#onSaveInstanceState`? Maybe we should add a `@CheckReturnValue` annotation so that it isn't missed.

Reply:
I just realized that `onSaveInstanceState(Bundle?)` does not return a value while used inside an `Activity`. We make have to introduce new functions to deal with this case and later split them into two different classes. Will take a stab at this.

Reply:
I investigated this and it is evident that we need a different delegate for the `Activity`

Reply:
@vinaysshenoy addressed this feedback. Can you plese review #888 first. So that I can re-request review on this PR. Thanks.